src/test/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionSpec.scala

```diff
@@ -0,0 +1,96 @@
+
+    val mockServer = RemoteServicesMockServer()
+
+    val credential: Credential = new MockGoogleCredential.Builder().build()
+    val googleServicesDAO: MockGoogleServicesDAO = new MockGoogleServicesDAO("test")
+    val executionServiceDAO = new HttpExecutionServiceDAO(mockServer.mockServerBaseUrl, mockServer.defaultWorkflowSubmissionTimeout)
+    val methodRepoDAO = new HttpMethodRepoDAO(mockServer.mockServerBaseUrl)
+  }
+
+  "WorkflowSubmission" should "get a batch of workflows" in withDefaultTestDatabase {
+    val workflowSubmission = new TestWorkflowSubmission(slickDataSource)
+
+    val workflowRecs = runAndWait(
+      for {
+        workflowRecs <- workflowQuery.findWorkflowsBySubmissionId(UUID.fromString(testData.submission1.submissionId)).take(workflowSubmission.batchSize).result
+        _ <- workflowQuery.batchUpdateStatus(workflowRecs, WorkflowStatuses.Queued)
+      } yield {
+        workflowRecs
+      }
+    )
+
+    assertResult(SubmitWorkflowBatch(workflowRecs.map(_.id))) {
+      Await.result(workflowSubmission.getUnlaunchedWorkflowBatch(), Duration.Inf)
+    }
+
+    assert(runAndWait(workflowQuery.findWorkflowByIds(workflowRecs.map(_.id)).result).forall(_.status == WorkflowStatuses.Launching.toString))
+  }
+
+  it should "schedule the next check when there are no workflows" in withDefaultTestDatabase {
+    val workflowSubmission = new TestWorkflowSubmission(slickDataSource)
+    assertResult(ScheduleNextWorkflowQuery) {
+      Await.result(workflowSubmission.getUnlaunchedWorkflowBatch(), Duration.Inf)
+    }
+  }
+
+  it should "have only 1 submission in a batch" in withDefaultTestDatabase {
+    val workflowSubmission = new TestWorkflowSubmission(slickDataSource, 100)
+
+    runAndWait(workflowQuery.batchUpdateStatus(WorkflowStatuses.Submitted, WorkflowStatuses.Queued))
+
+    Await.result(workflowSubmission.getUnlaunchedWorkflowBatch(), Duration.Inf) match {
+      case SubmitWorkflowBatch(workflowIds) =>
+        assertResult(1)(runAndWait(workflowQuery.findWorkflowByIds(workflowIds).map(_.submissionId).distinct.result).size)
+      case _ => fail("expected some workflows")
+    }
+  }
+
+  it should "submit a batch of workflows" in withDefaultTestDatabase {
+    val workflowSubmission = new TestWorkflowSubmission(slickDataSource)
+
+    withWorkspaceContext(testData.workspace) { ctx =>
+
+      Await.result( workflowSubmission.googleServicesDAO.storeToken(userInfo, UUID.randomUUID.toString), Duration.Inf )
+      val workflowIds = runAndWait(workflowQuery.getWithWorkflowIds(ctx, testData.submission1.submissionId)).map(_._1)
+
+      Await.result(workflowSubmission.submitWorkflowBatch(workflowIds), Duration.Inf) match {
+        case ScheduleNextWorkflowQuery =>
+          assert({
+            val workflowRecs = runAndWait(workflowQuery.filter(_.id inSetBind workflowIds).result)
+            workflowRecs.forall( _.status == "Submitted" )
+          }, "Not all workflows got submitted?")
+      }
```

Comment:
I would restructure this:
assertResult(ScheduleNextWorkflowQuery) { Await.result(workflowSubmission.submitWorkflowBatch(workflowIds), Duration.Inf) }
then do the following assert